src/UniGetUI/Pages/SoftwarePages/SoftwareUpdatesPage.cs

```diff
@@ -17,6 +18,88 @@
 
 namespace UniGetUI.Interface.SoftwarePages
 {
+    public class PauseTime
+    {
+        private int _daysTill;
+        public int Months { get { return Weeks / 4; } set { Weeks = value * 4; } }
+        public int Weeks { get { return Days / 7; } set { Days = value * 7; } }
+        public int Days { get { return _daysTill; } set { _daysTill = value; } }
+
+        public string GetDateFromNow()
+        {
+            DateTime NewTime = DateTime.Now.AddDays(_daysTill);
+            return NewTime.ToString("yyyy-MM-dd");
+        }
+
+        public void Parse(string Date)
+        {
+            try
+            {
+                DateTime ParsedDate = DateTime.ParseExact(Date, "yyyy-MM-dd", System.Globalization.CultureInfo.InvariantCulture);
+                DateTime Now = DateTime.Now;
+                if (ParsedDate > Now)
+                {
+                    _daysTill = (int)(ParsedDate - Now).TotalDays;
+                }
+                else
+                {
+                    _daysTill = (int)(Now - ParsedDate).TotalDays;
+                }
+            }
+            catch (FormatException ex)
+            {
+                Logger.Error($"Couldn't parse date {Date}:");
+                Logger.Error(ex);
+            }
+        }
+
+        public string StringRepresentation()
+        {
+            /*
```

Comment:
I reckon the body you left uncommented gets the work done. In fact, on the Settings Page, to set the "Check for updates interval" I do something really similar.